pydantic/_internal/_generate_schema.py

```diff
@@ -2158,10 +2202,27 @@ def inner_handler(obj: Any) -> CoreSchema:
             update_core_metadata(core_metadata, pydantic_js_annotation_functions=pydantic_js_annotation_functions)
         return _add_custom_serialization_from_json_encoders(self._config_wrapper.json_encoders, source_type, schema)
 
-    def _apply_single_annotation(self, schema: core_schema.CoreSchema, metadata: Any) -> core_schema.CoreSchema:
+    def _apply_single_annotation(
+        self,
+        schema: core_schema.CoreSchema,
+        metadata: Any,
+        check_unsupported_field_info_attributes: bool = True,
+    ) -> core_schema.CoreSchema:
         FieldInfo = import_cached_field_info()
 
         if isinstance(metadata, FieldInfo):
+            if check_unsupported_field_info_attributes and (
+                unsupported_attributes := self._get_unsupported_field_info_attributes(metadata)
+            ):
+                for unsupported_attr in unsupported_attributes:
```

Comment:
Can we do `for (attr, value) in unsupported_attributes` so we don't need `[0]` and `[1]`?